### PR TITLE
Cirrus: Script to manually prov. CI-like VMs

### PR DESCRIPTION
**Pull request by cevich**
Signed-off-by: Chris Evich <cevich@redhat.com>

### CODE REVIEW

**Comment by cevich** [comment]
Tested this manually and it works, but I don't think @nalind and @TomSweeneyRedHat have been exposed to this tool yet.  Mind giving it a run, make sure it works for you?   We use something very similar in libpod and it's super handy for debugging and working on CI-identical VMs in google.

**Comment by rhatdan** [comment]
When I run this, I get permission denied?
```
+  sudo podman run -it --rm -e AS_ID=3267 -e AS_USER=dwalsh --security-opt label=disable -v /tmp/get_ci_vm.sh_tmpdir_kTz4Mf:/home/dwalsh -v /home/dwalsh/.config/gcloud:/home/dwalsh/.config/gcloud -v /home/dwalsh/.config/gcloud/ssh:/home/dwalsh/.ssh -v /home/dwalsh/go/src/github.com/containers/storage:/home/dwalsh/go/src/github.com/containers/storage quay.io/cevich/gcloud_centos:latest --configuration=storage --project=storage-240716 compute instances create --zone=us-central1-b --image-project=libpod-218412 --image=fedora-cloud-base-30-1-2-1556821664 --custom-cpu=2 --custom-memory=4Gb --boot-disk-size=200 --labels=in-use-by=dwalsh dwalsh-fedora-cloud-base-30-1-2-1556821664
tar: ./.vagrant/machines/fedora/libvirt/private_key: Cannot open: Permission denied
                                                                                   tar: ./.vagrant/machines/debian/libvirt/private_key: Cannot open: Permission denied
```

**Comment by rhatdan** [comment]
BTW This is what I am executing

./hack/get_ci_vm.sh fedora-cloud-base-30-1-2-1556821664

**Comment by cevich** [comment]
Thanks Dan...Interesting, okay, I see where that's coming from.  I should probably have tar ignore everything matching a pattern in `.gitignore`.  I can't think of any reason for those items to get transferred to the VM.  Okay, fixing that...

**Comment by cevich** [comment]
okay, that will fix this failure mode, thanks to @nalind suggesting ``--exclude-vcs-ignores``.

**Comment by cevich** [comment]
@rhatdan thanks for testing this.  I added a note to the setup section, reminding new users to never set a ssh password.  Also added commit to fix git ignoring `cmd/container-storage' (oops, who did that?)

File: hack/get_ci_vm.sh [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    # If this is the only config, make it the default to avoid persistent warnings from gcloud
+    [[ -r "$HOME/.config/gcloud/configurations/config_default" ]] || \
+        ln "$HOME/.config/gcloud/configurations/config_storage" \
+           "$HOME/.config/gcloud/configurations/config_default"
+fi
+
+# Couldn't make rsync work with gcloud's ssh wrapper: ssh-keys generated on the fly
+TARBALL=$VMNAME.tar.bz2
+echo -e "\n${YEL}Packing up local repository into a tarball.${NOR}"
+showrun --background tar cjf $TMPDIR/$TARBALL --warning=no-file-changed --exclude=.git --exclude-vcs-ignores -C $STORAGEROOT .
```
Comment (rhatdan) [inline hack/get_ci_vm.sh]: This is wrong, do not --exclude=.git
We need git when we login the --exclude-vcs-ignores is enough.

Comment (cevich) [reply hack/get_ci_vm.sh]: We do this in libpod because it saves time in the transfer, but if it's inconvenient here I can strip it, np.

Comment (cevich) [reply hack/get_ci_vm.sh]: kk, got it, and `--force` pushed.

**Comment by cevich** [comment]
@rhatdan just for giggles, when including `.git` using the crappy bus WiFi:

```
cevich-fedora-cloud-base-29-1-2-1541789245.tar.bz2                     100%   63MB   1.7MB/s   00:38    
```

vs with `--exclude=.git`:

```
cevich-fedora-cloud-base-29-1-2-1541789245.tar.bz2                     100%  982KB 327.3KB/s   00:03    
```

(I rarely do this, so it's no biggie to include)

**Comment by rhatdan** [comment]
LGTM
Lets get this in. and I can continue testing.

**Comment by rhatdan** [comment]
@nalind @TomSweeneyRedHat @giuseppe @vrothberg PTAL

**Comment by giuseppe** [comment]
LGTM

**Comment by cevich** [comment]
Thanks guys, esp @rhatdan for oodles of patients weeding through all my duct-tape :smiley:

**Pull request merged**